### ISSUE TITLE
ZodError on experimental capability property (mcp inspector affected)

### ISSUE THREAD

**Issue by darena-mhaque** [modelcontextprotocol/typescript-sdk#1337]
**Describe the bug**
If the server returns a JSON response with capabilities as such:

```json
{
  "capabilities": {
    "experimental": {
      "hello": true
    }
  }
}
```

This results in the following error in the console:
<img width="237" height="191" alt="Image" src="https://github.com/user-attachments/assets/20b42fbb-edc3-458e-a0c5-9df0b6756e44" />

However, if the server responds with `hello` being an object instead, it works out:
```json
{
  "capabilities": {
    "experimental": {
      "hello": {
        "value": true
      }
    }
  }
}
```

The above does not throw an error.

**To Reproduce**
1. Use the mcp inspector
2. Create an mcp server that returns capabilities with experimental data in the initialize request like this:

```json
{
  "capabilities": {
    "experimental": {
      "hello": true
    }
  }
}
```

3. Connect to server
4. Observe error should be thrown and an error displayed in the mcp inspector.

**Expected behavior**
This should not throw an error as the value for each experimental flag can be any `object`, meaning it should be possible to provide strings, booleans, ints, etc. The code is currently taking `object` to mean literally.

I suspect the issue is [here in the zod schema](https://github.com/modelcontextprotocol/typescript-sdk/blob/v1.x/src/types.ts#L22)

```typescript
const AssertObjectSchema = z.custom<object>((v): v is object => v !== null && (typeof v === 'object' || typeof v === 'function'));
```

`typeof v` here could be `string`, `number`, etc.

**Logs**
N/A

**Additional context**
N/A

**Comment by darena-mhaque** [comment]
Hmm, this could be a misunderstanding of the specification. https://modelcontextprotocol.io/specification/2025-11-25/schema#servercapabilities

The specification above does indicate that the value should be an `object`:
```typescript
interface ServerCapabilities {
  experimental?: { [key: string]: object };
}
```

It doesn't explicitly say `any`, like some other areas of the specification, so maybe it is intended that the value be an actual object rather than a primitive type like `number` or `string`.

I think I'll close this ticket out for now.
